deployment/k8sPaiLibrary/maintainlib/update.py

```diff
@@ -0,0 +1,253 @@
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import sys
+import time
+import yaml
+import requests
+import logging
+import logging.config
+
+from . import add as k8s_add
+from . import clean as k8s_clean
+from . import common as k8s_common
+from . import remove as k8s_remove
+
+from ...confStorage import conf_storage_util
+from ...confStorage.download import download_configuration
+from ...paiLibrary.common import directory_handler
+from ...paiLibrary.common import kubernetes_handler
+from ...paiLibrary.clusterObjectModel import objectModelFactory
+
+
+class update:
+
+    def __init__(self, **kwargs):
+        self.logger = logging.getLogger(__name__)
+
+        self.kube_config_path = None
+        if "kube_config_path" in kwargs and kwargs[ "kube_config_path" ] != None:
+            self.kube_config_path = kwargs[ "kube_config_path" ]
+
+        if self.kube_config_path == None:
+            self.logger.error("Unable to find KUBECONFIG. Please ensure that you have passed the correct path.")
+            sys.exit(1)
+
+        self.time = str(int(time.time()))
+        self.tmp_path = "./tmp-machine-update-{0}".format(self.time)
+
+        self.k8s_configuration = None
+        self.node_list_from_k8s = None
+        self.node_config_from_cluster_conf = None
+        self.node_config_from_k8s = None
+
+
+
+    def get_latest_configuration_from_pai(self):
+
+        self.logger.info("Try to get latest configuration from kubernetes.")
+        directory_handler.directory_create(self.tmp_path)
+
+        config_get = download_configuration(config_output_path=self.tmp_path, kube_config_path=self.kube_config_path)
+        config_get.run()
+        objectModel = objectModelFactory.objectModelFactory(self.tmp_path)
+        ret = objectModel.objectModelPipeLine()
+
+        directory_handler.directory_delete(self.tmp_path)
+        self.logger.info("Successfully get latest configuration from Kubernetes.")
+        return ret["k8s"]
+
+
+
+    def get_node_list_from_k8s_api(self):
```

Comment:
I'd prefer the style likes `ClusterConfiguration.get_latest()`,  `ClusterConfiguration.get_node_list()`, the `from` indicate that we may be able to extract a object.
What do you think?

Reply:
```node_list``` comes from kubernetes api. So I don't think it should be part of the cluster configuration.
The obect you say, should be a future feature. Cluster object model. And now, we handle it as the easiest way. Just get all configuration from a dictionary.